azerothcore/azerothcore-wotlk
# refact(CORE/Mail): Avoid big mail loading and saving time by only loading 50 mails at a time.

## CHANGES PROPOSED:
-  Changed the way mail is handled to only have 50 mail loaded from the DB when the client request mail update to keep load time low and avoid crashes with big amount of mails.
-  Save changed mail when new mail is fetched as now the mail size is much lower and can be done without problems


## TESTS PERFORMED:
- Build on windows with no errors
- Created a character with 150K mails to test loading time when mail is opened
- Send mail between characters, from console and AH with no apparent errors



## HOW TO TEST THE CHANGES:

- Use the mail system as much as possible with different characters online and offline. Including AH.
- Send many mails to a character to check load time

## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
